slides/snippets/src/main/java/MergeText.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// [START slides_text_merging]
+import com.google.api.client.googleapis.auth.oauth2.GoogleCredential;
+import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.services.drive.Drive;
+import com.google.api.services.drive.model.File;
+import com.google.api.services.sheets.v4.Sheets;
+import com.google.api.services.sheets.v4.model.ValueRange;
+import com.google.api.services.slides.v1.Slides;
+import com.google.api.services.slides.v1.SlidesScopes;
+import com.google.api.services.slides.v1.model.*;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class MergeText {
+
+    // Create a new presentation
+    public static void mergeText(String templatePresentationId, String spreadsheetId) {
+        try {
+            // String templatePresentationId = "55BYRgExa5NEv9vMegpPBaWESy7oKTK21l4P2DcX5l4";
+            // String spreadsheetId = "1n6LkAHJ-rTflmy7L-BJil5KVKN3eECqrO3V1quv1QLk";
+
+            // Load user credentials from environment.
+            // See <some URL> for additional ways to acquire credentials for
+            // your application type and use case (CLI, web application, etc.)
+            List<String> scopes = Arrays.asList(SlidesScopes.PRESENTATIONS);
+            GoogleCredential credential = GoogleCredential.getApplicationDefault().createScoped(scopes);
```

Comment:
What would we do here for APIs that require a non-service account use? DWD? Would we embed the target user to impersonate in our test domain in the sample code?

Reply:
ADC isn't restricted to service accounts, the credentials can be tied to a user/refresh token. Regardless, auth is likely the hardest thing we need to decide as we're trying to pick the least bad option. User auth doesn't lend itself to isolated snippets. So far only options I've seen are:

* Have very complicated samples with user auth built in (like our quickstarts), but user auth is highly dependent on the app's platform/form factor and whatever we do will be wrong for many users.
* Add a layer of indirection and centralize getting credentials in another block of code. Has the same problem with form factors, indirection adds complexity too.
* Use ADC. Honestly, not very much different than the previous options as it's likely wrong for a majority of users. Only difference is the indirection is going towards a feature of the client lib, not elsewhere in the sample repo.

Curious to hear other approaches here.

Absent better options, idea here is to just show *something* simple to convey the need for credentials but couch it in a comment that makes it clear that this is just one way and you need to pick the correct mechanism for your app.

Reply:
SGTM, the best of bad options.